### PR TITLE
separate client and main dialog

### PR DESCRIPTION
**Pull request by jreidinger**
- [x] adapt yast2-storage-ng ( not needed, is independent )
- [x] test installation with new one ( not needed, is independent )
- [x] test runtime with this change

it requires change also in yast-storage-ng and testing it, if it works well.

### CODE REVIEW

File: src/lib/y2partitioner/clients/installation.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+module Y2Partitioner
+  module Clients
+    # Entry point for partitioner in installation. Difference is that it gets current
+    # device graph and returns as result modified or original device graph according to
+    # user decision.
+    class Installation
+      extend Yast::Logger
+
+      # Run the client
+      # @param allow_commit [Boolean] can we pass the point of no return
```
Comment (mvidner) [inline src/lib/y2partitioner/clients/installation.rb]: device_graph needs a description

**Comment by ancorgs** [comment]
The API of `Dialogs::Main` fits the installer needs. :+1:

So if this works as expected during installation (I'm not familiar with `CWM::Dialog`), then LGTM

**Comment by jreidinger** [comment]
@mvidner 

1. rubocop fixed
2. this pr do not change usage from storage-ng. It still have main client which is used previously. So it do not depend on each other. storage-ng can be modified later and according to @kobliha it can be done by team S.

**Comment by mvidner** [comment]
> 2. this pr ~~do not~~ *does not* change usage from storage-ng. It still ~~have~~ *has* main client which is used previously. So it ~~do not~~ *does not*  depend on each other. storage-ng can be modified later and according to kobliha it can be done by team S.

Thanks for the explanation. So you could have marked the item as [x] done, with "(not needed)"

**Pull request merged**